### ISSUE TITLE
use `key:value` struct tags?

### ISSUE THREAD

**Issue by fortytw2** [alexflint/go-restructure#2]
I think that

```go
type Example struct {
    MyField string `regex:"[^@]+"`
}
```
instead of 
```go
type Example struct {
    MyField string `[^@]+`
}
```

would mesh more with other libraries, standard style, etc. Worth a discussion (awesome library, btw)

**Comment by sethgrid** [comment]
`go vet` will complain if the tag is not `key:"value"`:
https://github.com/golang/tools/blob/master/cmd/vet/structtag.go#L70

**Comment by telotortium** [comment]
You should probably use the tag "restructure" instead, since that's the name of your library.

**Comment by renstrom** [comment]
I think `re` (after Python's regexp package) is nice and succinct tag name.

```go
type Example struct {
    MyField string `re:"[^@]+"`
}
```

**Comment by cathalgarvey** [comment]
Would *really* love this feature. Perhaps use `restructure` OR `re`? If it has to be one, propriety suggests it should be `restructure`, because explicit is better than...

The only problem with using 'normal' tags is that it imposes a backslash burden on quotes in the regex, but the advantages, IMO, outweigh that. The convenience of being able to Regex-out something and immediately serialise to JSON with tagged params, for example, is an obvious use-case.

**Comment by alexflint** [comment]
This was resolved in #12
